### PR TITLE
Avoid unnecessary QString heap allocations

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

File: src/library/basesqltablemodel.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@ const int kMaxSortColumns = 3;
 // Constant for getModelSetting(name)
 const QString COLUMNS_SORTING = QStringLiteral("ColumnsSorting");
 
-const QString kEmptyString = QStringLiteral("");
+const QString kEmptyString = QLatin1String("");
```
Comment (daschuer) [inline src/library/basesqltablemodel.cpp:38]: This constant is useless and can be removed.

File: src/preferences/dialog/dlgpreflv2.cpp [hunk 2]
```diff
@@ -61,7 +61,7 @@ DlgPrefLV2::DlgPrefLV2(QWidget* pParent, LV2Backend* lv2Backend,
         button->setProperty("id", QVariant(pEffectManifest->id()));
         connect(button, SIGNAL(clicked()), this, SLOT(slotDisplayParameters()));
     }
-    effectNameLabel->setText(QStringLiteral(""));
+    effectNameLabel->setText(QLatin1String(""));
```
Comment (daschuer) [inline src/preferences/dialog/dlgpreflv2.cpp:64]:
```suggestion
    effectNameLabel->clear());
```

File: src/widget/wsearchlineedit.cpp [hunk 3]
```diff
@@ -19,7 +19,7 @@ const mixxx::Logger kLogger("WSearchLineEdit");
 
 const QColor kDefaultBackgroundColor = QColor(0, 0, 0);
 
-const QString kEmptySearch = QStringLiteral("");
+const QString kEmptySearch = QLatin1String("");
```
Comment (daschuer) [inline src/widget/wsearchlineedit.cpp:22]: Can be removed completely

File: src/library/dao/cuedao.cpp [hunk 4]
```diff
@@ -20,7 +20,7 @@ namespace {
 const mixxx::Logger kLogger = mixxx::Logger("CueDAO");
 
 // The label column is not nullable!
-const QVariant kEmptyLabel = QVariant(QStringLiteral(""));
+const QVariant kEmptyLabel = QVariant(QLatin1String(""));
```
Comment (daschuer) [inline src/library/dao/cuedao.cpp:23]: Please remove.

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/widget/wsearchlineedit.cpp (hunk 3 above)
Comment (Holzhaus) [reply src/widget/wsearchlineedit.cpp]: I was confused why this explicitly checks that the search string is not null, but there is not comment why this is necessary. I removed it and noticed not regression, but please give it a long, hard stare.

**Review by daschuer — APPROVED** [review]
Thank you. LGTM.

**Comment by daschuer** [comment]
Waiting for CI

**Comment by uklotzde** [comment]
Next one ;) LGTM, too.

**Pull request merged**